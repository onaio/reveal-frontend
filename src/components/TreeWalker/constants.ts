export const ACTIVE = 'Active';
export const FEATURE = 'Feature';
export const FIND_BY_ID = 'location/findByJurisdictionIds';
export const FIND_BY_PROPERTIES = 'location/findByProperties';
export const LOCATION = 'location';
export const COULDNT_LOAD_CHILDREN = 'Could not load children';
export const COULDNT_LOAD_PARENTS = 'Could not load parents';
export const NOT_AVAILABLE = 'N/A';
