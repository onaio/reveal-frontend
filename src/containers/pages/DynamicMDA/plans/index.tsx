import reducerRegistry from '@onaio/redux-reducer-registry';
import React from 'react';
import { connect } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { Store } from 'redux';
import {
  HIDDEN_PLAN_STATUSES,
  SUPERSET_DYNAMIC_MDA_REPORTING_PLANS_SLICE,
} from '../../../../configs/env';
import { MDA_PLANS } from '../../../../configs/lang';
import { QUERY_PARAM_TITLE, REPORT_MDA_PLAN_URL } from '../../../../constants';
import { getPlanStatusToDisplay, getQueryParams, RouteParams } from '../../../../helpers/utils';
import supersetFetch from '../../../../services/superset';
import GenericPLansReducer, {
  genericFetchPlans,
  makeGenericPlansArraySelector,
  reducerName as GenericPLansReducerName,
} from '../../../../store/ducks/generic/plans';
import { GenericPlan } from '../../../../store/ducks/generic/plans';
import { InterventionType } from '../../../../store/ducks/plans';
import { GenericPlanListProps, GenericPlansList } from '../../GenericPlansList';

/** register the MDA plan definitions reducer */
reducerRegistry.register(GenericPLansReducerName, GenericPLansReducer);

/** a list of plan statuses to be displayed */
const allowedPlanStatusList = getPlanStatusToDisplay(HIDDEN_PLAN_STATUSES);

/** selector for Dynamic MDA plans */
const MDAPlansArraySelector = makeGenericPlansArraySelector();

/** Simple component that loads a list of MDA plans */
const MDAPLansList = (props: GenericPlanListProps & RouteComponentProps) => {
  return (
    <div>
      <GenericPlansList {...props} />
    </div>
  );
};

/** Declare default props for MDAPLansList */
const defaultProps: GenericPlanListProps = {
  fetchPlans: genericFetchPlans,
  pageTitle: MDA_PLANS,
  pageUrl: REPORT_MDA_PLAN_URL,
  plans: [],
  service: supersetFetch,
  supersetReportingSlice: SUPERSET_DYNAMIC_MDA_REPORTING_PLANS_SLICE,
};

MDAPLansList.defaultProps = defaultProps;

export { MDAPLansList };

/** Connect the component to the store */

/** interface to describe props from mapStateToProps */
interface DispatchedStateProps extends RouteComponentProps<RouteParams> {
  plans: GenericPlan[];
}

/** map state to props */
const mapStateToProps = (
  state: Partial<Store>,
  ownProps: RouteComponentProps<RouteParams>
): DispatchedStateProps => {
  const searchedTitle = getQueryParams(ownProps.location)[QUERY_PARAM_TITLE] as string;
  const plans = MDAPlansArraySelector(state, {
    interventionTypes: [InterventionType.DynamicMDA],
    plan_title: searchedTitle,
    statusList: allowedPlanStatusList,
  });

  return {
    ...ownProps,
    plans,
  };
};

/** map dispatch to props */
const mapDispatchToProps = { fetchPlans: genericFetchPlans };

/** Connected MDA Plans component */
const ConnectedMDAPLansList = connect(mapStateToProps, mapDispatchToProps)(MDAPLansList);

export default ConnectedMDAPLansList;
