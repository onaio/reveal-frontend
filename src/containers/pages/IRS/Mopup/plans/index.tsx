import React from 'react';
import { RouteComponentProps } from 'react-router-dom';
import { MOP_UP_REPORTING_TITLE } from '../../../../../configs/lang';
import { IRS_MOP_UP_REPORT_URL } from '../../../../../constants';
import { RouteParams } from '../../../../../helpers/utils';
import ConnectedIRSPlansList from '../../plans';

export const IRSMopUpReporting = (props: RouteComponentProps<RouteParams>) => {
  const plansProps = {
    ...props,
    pageTitle: MOP_UP_REPORTING_TITLE,
    pageUrl: IRS_MOP_UP_REPORT_URL,
  };
  return <ConnectedIRSPlansList {...plansProps} />;
};
