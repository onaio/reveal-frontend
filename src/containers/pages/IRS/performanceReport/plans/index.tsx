import React from 'react';
import { RouteComponentProps } from 'react-router-dom';
import { IRS_PERFORMANCE_REPORTING_TITLE } from '../../../../../configs/lang';
import { PERFORMANCE_REPORT_IRS_PLAN_URL } from '../../../../../constants';
import { RouteParams } from '../../../../../helpers/utils';
import ConnectedIRSPlansList from '../../plans';

export const IRSPerfomenceReport = (props: RouteComponentProps<RouteParams>) => {
  const plansProps = {
    ...props,
    pageTitle: IRS_PERFORMANCE_REPORTING_TITLE,
    pageUrl: PERFORMANCE_REPORT_IRS_PLAN_URL,
  };
  return <ConnectedIRSPlansList {...plansProps} />;
};
