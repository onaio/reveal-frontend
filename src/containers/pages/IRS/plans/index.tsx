import reducerRegistry from '@onaio/redux-reducer-registry';
import React from 'react';
import { connect } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { Store } from 'redux';
import { HIDDEN_PLAN_STATUSES, SUPERSET_IRS_REPORTING_PLANS_SLICE } from '../../../../configs/env';
import { IRS_PLANS } from '../../../../configs/lang';
import { QUERY_PARAM_TITLE, REPORT_IRS_PLAN_URL } from '../../../../constants';
import { getPlanStatusToDisplay, getQueryParams, RouteParams } from '../../../../helpers/utils';
import supersetFetch from '../../../../services/superset';
import GenericPLansReducer, {
  genericFetchPlans,
  GenericPlan,
  makeGenericPlansArraySelector,
  reducerName as GenericPLansReducerName,
} from '../../../../store/ducks/generic/plans';
import { InterventionType } from '../../../../store/ducks/plans';
import { GenericPlanListProps, GenericPlansList } from '../../GenericPlansList';

/** register the IRS plan definitions reducer */
reducerRegistry.register(GenericPLansReducerName, GenericPLansReducer);

/** a list of plan statuses to be displayed */
const allowedPlanStatusList = getPlanStatusToDisplay(HIDDEN_PLAN_STATUSES);

/** selector for IRS plans */
const makeIRSPlansArraySelector = makeGenericPlansArraySelector();

/** Simple component that loads a preview list of IRS plans */
const IRSPlansList = (props: GenericPlanListProps & RouteComponentProps) => {
  return (
    <div>
      <GenericPlansList {...props} />
    </div>
  );
};

/** Declare default props for IRSPlansList */
const defaultProps: GenericPlanListProps = {
  fetchPlans: genericFetchPlans,
  pageTitle: IRS_PLANS,
  pageUrl: REPORT_IRS_PLAN_URL,
  plans: [],
  service: supersetFetch,
  supersetReportingSlice: SUPERSET_IRS_REPORTING_PLANS_SLICE,
};

IRSPlansList.defaultProps = defaultProps;

export { IRSPlansList };

/** Connect the component to the store */

/** interface to describe props from mapStateToProps */
interface DispatchedStateProps extends RouteComponentProps<RouteParams> {
  plans: GenericPlan[];
}

/** map state to props */
const mapStateToProps = (
  state: Partial<Store>,
  ownProps: RouteComponentProps<RouteParams>
): DispatchedStateProps => {
  const searchedTitle = getQueryParams(ownProps.location)[QUERY_PARAM_TITLE] as string;
  const plans = makeIRSPlansArraySelector(state, {
    interventionTypes: [InterventionType.IRS, InterventionType.DynamicIRS],
    plan_title: searchedTitle,
    statusList: allowedPlanStatusList,
  });

  return {
    ...ownProps,
    plans,
  };
};

/** map dispatch to props */
const mapDispatchToProps = { fetchPlans: genericFetchPlans };

/** Connected ActiveFI component */
const ConnectedIRSPlansList = connect(mapStateToProps, mapDispatchToProps)(IRSPlansList);

export default ConnectedIRSPlansList;
