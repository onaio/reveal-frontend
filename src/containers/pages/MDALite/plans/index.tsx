import reducerRegistry from '@onaio/redux-reducer-registry';
import React from 'react';
import { connect } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { Store } from 'redux';
import {
  HIDDEN_PLAN_STATUSES,
  SUPERSET_MDA_LITE_REPORTING_PLANS_SLICE,
} from '../../../../configs/env';
import { MDA_LITE_PLANS } from '../../../../configs/lang';
import { QUERY_PARAM_TITLE, REPORT_MDA_LITE_PLAN_URL } from '../../../../constants';
import { getPlanStatusToDisplay, getQueryParams, RouteParams } from '../../../../helpers/utils';
import supersetFetch from '../../../../services/superset';
import GenericPlansReducer, {
  genericFetchPlans,
  GenericPlan,
  makeGenericPlansArraySelector,
  reducerName as genericReducerName,
} from '../../../../store/ducks/generic/plans';
import { InterventionType } from '../../../../store/ducks/plans';
import { GenericPlanListProps, GenericPlansList } from '../../GenericPlansList';

/** register the MDA plan definitions reducer */
reducerRegistry.register(genericReducerName, GenericPlansReducer);

/** a list of plan statuses to be displayed */
const allowedPlanStatusList = getPlanStatusToDisplay(HIDDEN_PLAN_STATUSES);

/** selector for MDA lite plans */
const makeMDALitePlansArraySelector = makeGenericPlansArraySelector();

/** Simple component that loads a preview list of MDA plans */
const MDALitePlansList = (props: GenericPlanListProps & RouteComponentProps) => {
  return (
    <div>
      <GenericPlansList {...props} />
    </div>
  );
};

/** Declare default props for MDALitePlansList */
const defaultProps: GenericPlanListProps = {
  fetchPlans: genericFetchPlans,
  pageTitle: MDA_LITE_PLANS,
  pageUrl: REPORT_MDA_LITE_PLAN_URL,
  plans: [],
  service: supersetFetch,
  supersetReportingSlice: SUPERSET_MDA_LITE_REPORTING_PLANS_SLICE,
};

MDALitePlansList.defaultProps = defaultProps;

export { MDALitePlansList };

/** Connect the component to the store */

/** interface to describe props from mapStateToProps */
interface DispatchedStateProps extends RouteComponentProps<RouteParams> {
  plans: GenericPlan[];
}

/** map state to props */
const mapStateToProps = (
  state: Partial<Store>,
  ownProps: RouteComponentProps<RouteParams>
): DispatchedStateProps => {
  const searchedTitle = getQueryParams(ownProps.location)[QUERY_PARAM_TITLE] as string;
  const MDALitePlansArray = makeMDALitePlansArraySelector(state, {
    interventionTypes: [InterventionType.MDALite],
    plan_title: searchedTitle,
    statusList: allowedPlanStatusList,
  });

  return {
    ...ownProps,
    plans: MDALitePlansArray,
  };
};

/** map dispatch to props */
const mapDispatchToProps = { fetchPlans: genericFetchPlans };

/** Connected ActiveFI component */
const ConnectedMDALitePlansList = connect(mapStateToProps, mapDispatchToProps)(MDALitePlansList);

export default ConnectedMDALitePlansList;
