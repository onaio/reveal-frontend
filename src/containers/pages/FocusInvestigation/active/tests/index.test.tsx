import { library } from '@fortawesome/fontawesome-svg-core';
import { faExternalLinkSquareAlt } from '@fortawesome/free-solid-svg-icons';
import { renderTable } from '@onaio/drill-down-table';
import reducerRegistry from '@onaio/redux-reducer-registry';
import superset from '@onaio/superset-connector';
import { mount, shallow } from 'enzyme';
import { createBrowserHistory } from 'history';
import { cloneDeep } from 'lodash';
import MockDate from 'mockdate';
import React from 'react';
import { Helmet } from 'react-helmet';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import { CURRENT_FOCUS_INVESTIGATION } from '../../../../../configs/lang';
import { FI_URL, REACTIVE_QUERY_PARAM, ROUTINE_QUERY_PARAM } from '../../../../../constants';
import store from '../../../../../store';
import * as planDefFixtures from '../../../../../store/ducks/opensrp/PlanDefinition/tests/fixtures';
import { fetchPlansByUser } from '../../../../../store/ducks/opensrp/planIdsByUser';
import reducer, {
  fetchPlans,
  reducerName,
  removePlansAction,
} from '../../../../../store/ducks/plans';
import { InterventionType } from '../../../../../store/ducks/plans';
import * as fixtures from '../../../../../store/ducks/tests/fixtures';
import ConnectedActiveFocusInvestigation, { ActiveFocusInvestigation } from '../../active';
import { activeFocusInvestigationProps, selectedPlan1, selectedPlan24 } from './fixtures';

reducerRegistry.register(reducerName, reducer);

library.add(faExternalLinkSquareAlt);
const history = createBrowserHistory();
jest.mock('../../../../../configs/env');
jest.mock('../../../../../helpers/dataLoading/plans', () => {
  return {
    loadPlansByUserFilter: async () => [],
  };
});

describe('containers/pages/ActiveFocusInvestigation', () => {
  beforeEach(() => {
    jest.resetAllMocks();
    store.dispatch(removePlansAction);
    MockDate.reset();
  });

  it('renders without crashing', () => {
    const mock: any = jest.fn();
    const props = {
      caseTriggeredPlans: [fixtures.plan2],
      fetchPlansActionCreator: jest.fn(),
      history,
      location: mock,
      match: mock,
      routinePlans: [fixtures.plan1],
    };
    shallow(
      <Router history={history}>
        <ActiveFocusInvestigation {...props} />
      </Router>
    );
  });

  it('renders without crashing for null jurisdictions_name_path', () => {
    const mock: any = jest.fn();
    const props = {
      caseTriggeredPlans: [fixtures.plan2],
      fetchPlansActionCreator: jest.fn(),
      history,
      location: mock,
      match: mock,
      routinePlans: [fixtures.plan1],
    };
    shallow(
      <Router history={history}>
        <ActiveFocusInvestigation {...props} />
      </Router>
    );
  });

  it('renders ActiveFocusInvestigation correctly $ changes page title', async () => {
    const mock: any = jest.fn();
    mock.mockImplementation(() => Promise.resolve(fixtures.plans));
    const props = {
      caseTriggeredPlans: [fixtures.plan2],
      fetchPlansActionCreator: jest.fn(),
      history,
      location: mock,
      match: mock,
      routinePlans: [fixtures.plan1],
      supersetService: mock,
    };
    const wrapper = mount(
      <Router history={history}>
        <ActiveFocusInvestigation {...props} />
      </Router>
    );

    await new Promise(resolve => setImmediate(resolve));
    wrapper.update();
    const helmet = Helmet.peek();
    expect(helmet.title).toEqual(CURRENT_FOCUS_INVESTIGATION);
    expect(wrapper.find(ActiveFocusInvestigation).props().caseTriggeredPlans).toEqual(
      activeFocusInvestigationProps.caseTriggeredPlans
    );
    expect(wrapper.find(ActiveFocusInvestigation).props().routinePlans).toEqual(
      activeFocusInvestigationProps.routinePlans
    );
    expect(wrapper.find('HeaderBreadcrumb').length).toEqual(1);
    expect(wrapper.find('HeaderBreadcrumb').props()).toEqual({
      currentPage: {
        label: 'Focus Investigations',
        url: '/focus-investigation',
      },
      pages: [
        {
          label: 'Home',
          url: '/',
        },
      ],
    });
    wrapper.unmount();
  });

  it('renders ActiveFocusInvestigation correctly for null jurisdictions', async () => {
    const mock: any = jest.fn();
    mock.mockImplementation(() => Promise.resolve(fixtures.plans));
    const props = {
      caseTriggeredPlans: [fixtures.plan2],
      fetchPlansActionCreator: jest.fn(),
      history,
      location: mock,
      match: {
        isExact: true,
        params: {
          jurisdiction_parent_id: '2977',
          plan_id: fixtures.plan2.id,
        },
        path: FI_URL,
        url: FI_URL,
      },
      plan: fixtures.plan2,
      routinePlans: [fixtures.plan1],
      supersetService: mock,
    };
    const wrapper = mount(
      <Router history={history}>
        <ActiveFocusInvestigation {...props} />
      </Router>
    );
    await new Promise(resolve => setImmediate(resolve));
    wrapper.update();
    expect(wrapper.find(ActiveFocusInvestigation).props().caseTriggeredPlans).toEqual(
      activeFocusInvestigationProps.caseTriggeredPlans
    );
    expect(wrapper.find(ActiveFocusInvestigation).props().routinePlans).toEqual(
      activeFocusInvestigationProps.routinePlans
    );
    wrapper.unmount();
  });

  it('works with the Redux store', async () => {
    const envModule = require('../../../../../configs/env');
    envModule.ENABLED_PLAN_TYPES = 'FI,IRS,MDA,MDA-Point,Dynamic-FI,Dynamic-IRS,Dynamic-MDA'.split(
      ','
    );
    store.dispatch(fetchPlans([...fixtures.plans, fixtures.plan104]));
    const mock: any = jest.fn();
    mock.mockImplementation(() => Promise.resolve([...fixtures.plans, fixtures.plan104]));
    const props = {
      history,
      location: mock,
      match: mock,
      supersetService: mock,
    };
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedActiveFocusInvestigation {...props} />
        </Router>
      </Provider>
    );
    await new Promise(resolve => setImmediate(resolve));
    wrapper.update();
    expect(wrapper.text()).toMatchSnapshot('A large unstyled string of the rendered output');
    wrapper.find('DrillDownTable').forEach(table => renderTable(table));
  });

  it('calls superset with the correct params', async () => {
    const actualFormData = superset.getFormData;
    const getFormDataMock: any = jest.fn();
    getFormDataMock.mockImplementation((...args: any) => {
      return actualFormData(...args);
    });
    superset.getFormData = getFormDataMock;
    const mock: any = jest.fn();
    const supersetMock: any = jest.fn();
    supersetMock.mockImplementation(() => Promise.resolve(fixtures.plans));
    const props = {
      history,
      location: mock,
      match: mock,
      supersetService: supersetMock,
    };
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedActiveFocusInvestigation {...props} />
        </Router>
      </Provider>
    );
    await new Promise(resolve => setImmediate(resolve));
    wrapper.update();
    const supersetParams = {
      adhoc_filters: [
        {
          clause: 'WHERE',
          comparator: 'FI',
          expressionType: 'SIMPLE',
          operator: '==',
          subject: 'plan_intervention_type',
        },
      ],
      row_limit: 2000,
    };

    const supersetCallList = [
      [
        2000,
        [{ comparator: InterventionType.FI, operator: '==', subject: 'plan_intervention_type' }],
      ],
    ];
    expect((superset.getFormData as any).mock.calls).toEqual(supersetCallList);
    expect(supersetMock).toHaveBeenCalledWith(0, supersetParams);
    wrapper.unmount();
  });

  it('does not show loading when we have resolved promises', async () => {
    // resolve superset's request with empty data, it should not show the loader.
    const mock: any = jest.fn();
    const supersetMock: any = jest.fn(async () => []);
    const props = {
      history,
      location: mock,
      match: mock,
      supersetService: supersetMock,
    };
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedActiveFocusInvestigation {...props} />
        </Router>
      </Provider>
    );
    // shows ripple here
    expect(wrapper.find('Ripple').length).toEqual(1);
    await new Promise(resolve => setImmediate(resolve));
    wrapper.update();
    // now you dont see the ripple
    expect(wrapper.find('Ripple').length).toEqual(0);
    wrapper.unmount();
  });

  it('handles search correctly for case triggered plans', async () => {
    store.dispatch(fetchPlans([fixtures.plan24, fixtures.plan25]));

    const props = {
      history,
      location: {
        pathname: FI_URL,
        search: `?${REACTIVE_QUERY_PARAM}=Jane`,
      },
      match: {
        isExact: true,
        params: {},
        path: `${FI_URL}`,
        url: `${FI_URL}`,
      },
      supersetService: jest.fn().mockImplementationOnce(() => Promise.resolve([])),
    };
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedActiveFocusInvestigation {...props} />
        </Router>
      </Provider>
    );
    renderTable(wrapper, 'Expect single non header tr with title includes jane');
  });

  it('handles case insensitive searches correctly', async () => {
    store.dispatch(fetchPlans([fixtures.plan1, fixtures.plan22]));

    const props = {
      history,
      location: {
        pathname: FI_URL,
        search: `?${ROUTINE_QUERY_PARAM}=LUANG`,
      },
      match: {
        isExact: true,
        params: {},
        path: `${FI_URL}`,
        url: `${FI_URL}`,
      },
      supersetService: jest.fn().mockImplementationOnce(() => Promise.resolve([])),
    };
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedActiveFocusInvestigation {...props} />
        </Router>
      </Provider>
    );
    await new Promise(resolve => setImmediate(resolve));
    wrapper.update();
    renderTable(wrapper, 'expect the luang tr as only non header tr');
    wrapper.unmount();
  });

  it('renders empty tables if search query does not match any case trigger or routine plans', async () => {
    store.dispatch(fetchPlans([fixtures.plan1, fixtures.plan22, fixtures.plan24, fixtures.plan25]));

    const props = {
      history,
      location: {
        pathname: FI_URL,
        search: `?${ROUTINE_QUERY_PARAM}=Amazon&${REACTIVE_QUERY_PARAM}=Amazon`,
      },
      match: {
        isExact: true,
        params: {},
        path: `${FI_URL}`,
        url: `${FI_URL}`,
      },
      supersetService: jest.fn().mockImplementationOnce(() => Promise.resolve([])),
    };
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedActiveFocusInvestigation {...props} />
        </Router>
      </Provider>
    );
    await new Promise(resolve => setImmediate(resolve));
    wrapper.update();
    renderTable(wrapper, 'Should only have header rows');
    expect(wrapper.text()).toMatchSnapshot('should have 2 no data found texts');
  });

  it('filters plans by userName resulting in no plans', async () => {
    const planDef1 = cloneDeep(planDefFixtures.plans[0]);
    planDef1.identifier = '10f9e9fa-ce34-4b27-a961-72fab5206ab6';
    const userName = 'ghost';
    store.dispatch(fetchPlans([fixtures.completeRoutinePlan, fixtures.plan2]));
    store.dispatch(fetchPlansByUser([planDef1], userName));
    const props = {
      history,
      location: {
        pathname: FI_URL,
        search: `?user=${userName}`,
      },
      match: {
        isExact: true,
        params: {},
        path: `${FI_URL}`,
        url: `${FI_URL}`,
      },
      supersetService: jest.fn().mockImplementationOnce(() => Promise.resolve([])),
    };
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedActiveFocusInvestigation {...props} />
        </Router>
      </Provider>
    );
    await new Promise(resolve => setImmediate(resolve));
    wrapper.update();
    expect(
      wrapper
        .find('Table')
        .at(0)
        .prop('data')
    ).toEqual([]);
    expect(
      wrapper
        .find('Table')
        .at(1)
        .prop('data')
    ).toEqual([]);

    // no plans but loader is not showing
    expect(wrapper.find('Ripple').length).toEqual(0);
  });

  it('filters plans by userName', async () => {
    const planDef1 = cloneDeep(planDefFixtures.plans[0]);
    planDef1.identifier = '10f9e9fa-ce34-4b27-a961-72fab5206ab6';
    const userName = 'ghost';
    store.dispatch(
      fetchPlans([fixtures.plan1, fixtures.completeRoutinePlan, fixtures.plan24, fixtures.plan2])
    );
    store.dispatch(fetchPlansByUser([planDef1], userName));
    const props = {
      history,
      location: {
        pathname: FI_URL,
        search: `?user=${userName}`,
      },
      match: {
        isExact: true,
        params: {},
        path: `${FI_URL}`,
        url: `${FI_URL}`,
      },
      supersetService: jest.fn().mockImplementationOnce(() => Promise.resolve([])),
    };
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedActiveFocusInvestigation {...props} />
        </Router>
      </Provider>
    );
    await new Promise(resolve => setImmediate(resolve));
    wrapper.update();

    expect(
      wrapper
        .find('Table')
        .at(0)
        .prop('data')
    ).toEqual([selectedPlan24]);
    expect(
      wrapper
        .find('Table')
        .at(1)
        .prop('data')
    ).toEqual([selectedPlan1]);
  });

  it('should handle falsy result correctly', async () => {
    const supersetServiceMock: any = jest
      .fn(() => Promise.resolve(fixtures.plans))
      .mockImplementationOnce(async () => null);

    const mock: any = jest.fn();
    const props = {
      caseTriggeredPlans: [fixtures.plan2],
      fetchPlansActionCreator: jest.fn(),
      history,
      location: mock,
      match: mock,
      routinePlans: [fixtures.plan1],
      supersetService: supersetServiceMock,
    };
    const wrapper = mount(
      <Router history={history}>
        <ActiveFocusInvestigation {...props} />
      </Router>
    );
    // shows ripple here
    expect(wrapper.find('Ripple').length).toEqual(1);

    await new Promise(resolve => setImmediate(resolve));
    wrapper.update();
    expect(props.fetchPlansActionCreator).not.toHaveBeenCalled();

    // now you dont see the ripple
    expect(wrapper.find('Ripple').length).toEqual(0);
    wrapper.unmount();
  });
});
