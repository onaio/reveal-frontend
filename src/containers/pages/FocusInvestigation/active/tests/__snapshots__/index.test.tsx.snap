// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/pages/ActiveFocusInvestigation handles case insensitive searches correctly: expect the luang tr as only non header tr tr index 0 1`] = `"NameFI StatusProvinceDistrictCantonVillageFocus AreaStatusCase Notif. DateCase Class."`;

exports[`containers/pages/ActiveFocusInvestigation handles case insensitive searches correctly: expect the luang tr as only non header tr tr index 1 1`] = `"NameFI StatusProvinceDistrictCantonVillageFocus AreaStatusStart DateEnd Date"`;

exports[`containers/pages/ActiveFocusInvestigation handles case insensitive searches correctly: expect the luang tr as only non header tr tr index 2 1`] = `"A1-Tha Luang Village 1 Focus 01 activeChadizaNaviluliNVI_439  A12019-07-312019-06-18"`;

exports[`containers/pages/ActiveFocusInvestigation renders empty tables if search query does not match any case trigger or routine plans: Should only have header rows tr index 0 1`] = `"NameFI StatusProvinceDistrictCantonVillageFocus AreaStatusCase Notif. DateCase Class."`;

exports[`containers/pages/ActiveFocusInvestigation renders empty tables if search query does not match any case trigger or routine plans: Should only have header rows tr index 1 1`] = `"NameFI StatusProvinceDistrictCantonVillageFocus AreaStatusStart DateEnd Date"`;

exports[`containers/pages/ActiveFocusInvestigation renders empty tables if search query does not match any case trigger or routine plans: should have 2 no data found texts 1`] = `"HomeFocus InvestigationsCurrent Focus InvestigationsReactiveFilterFilteruserSelect usernameRow Height Row HeightdefaultshorttallCustomize ColumnsCustomize ColumnsSelect columns to be displayedNameFI StatusProvinceDistrictCantonVillageFocus AreaStatusCase Notif. DateCase Class.Rows to display10203050PreviousPage    Of 0Next NameFI StatusProvinceDistrictCantonVillageFocus AreaStatusCase Notif. DateCase Class.No Investigations FoundRows to display10203050PreviousPage    Of 0Next RoutineAdd Focus InvestigationFilterFilteruserSelect usernameRow Height Row HeightdefaultshorttallCustomize ColumnsCustomize ColumnsSelect columns to be displayedNameFI StatusProvinceDistrictCantonVillageFocus AreaStatusStart DateEnd DateRows to display10203050PreviousPage    Of 0Next NameFI StatusProvinceDistrictCantonVillageFocus AreaStatusStart DateEnd DateNo Investigations FoundRows to display10203050PreviousPage    Of 0Next DefinitionsA1ActiveIndigenous case recorded within the last year.A2Residual Non-ActiveNo indigenous case during the last year, but withing the last 3 years.B1Cleared ReceptiveReceptive area but no indigenous cases within the last 3 years.B2Cleared Non-ReceptiveNon-receptive area."`;

exports[`containers/pages/ActiveFocusInvestigation works with the Redux store:  tr index 0 1`] = `"NameFI StatusProvinceDistrictCantonVillageFocus AreaStatusCase Notif. DateCase Class."`;

exports[`containers/pages/ActiveFocusInvestigation works with the Redux store:  tr index 0 2`] = `"NameFI StatusProvinceDistrictCantonVillageFocus AreaStatusStart DateEnd Date"`;

exports[`containers/pages/ActiveFocusInvestigation works with the Redux store:  tr index 1 1`] = `"A1-Tha Luang Village 1 Focus 01 completeCanton Tha LuangTha Luang VillageTLv1_02  A12019-06-18"`;

exports[`containers/pages/ActiveFocusInvestigation works with the Redux store:  tr index 1 2`] = `"A1-Tha Luang Village 1 Focus 01 activeChadizaNaviluliNVI_439  A12019-07-312019-06-18"`;

exports[`containers/pages/ActiveFocusInvestigation works with the Redux store:  tr index 2 1`] = `"A1-Tha Luang Village 1 Focus 01 completeCanton Tha LuangTha Luang VillageTLv1_02  A12019-07-312019-06-18"`;

exports[`containers/pages/ActiveFocusInvestigation works with the Redux store:  tr index 3 1`] = `"plan104 activeTrat 2Mueng Trat District 2Laem Klat Canton 2Laem Klat Moo 8Tha Sen 8  A12019-06-182019-07-30"`;

exports[`containers/pages/ActiveFocusInvestigation works with the Redux store: A large unstyled string of the rendered output 1`] = `"HomeFocus InvestigationsCurrent Focus InvestigationsReactiveFilterFilteruserSelect usernameRow Height Row HeightdefaultshorttallCustomize ColumnsCustomize ColumnsSelect columns to be displayedNameFI StatusProvinceDistrictCantonVillageFocus AreaStatusCase Notif. DateCase Class.Rows to display10203050PreviousPage    Of 1Next NameFI StatusProvinceDistrictCantonVillageFocus AreaStatusCase Notif. DateCase Class.A1-Tha Luang Village 1 Focus 01 completeCanton Tha LuangTha Luang VillageTLv1_02  A12019-06-18Rows to display10203050PreviousPage    Of 1Next RoutineAdd Focus InvestigationFilterFilteruserSelect usernameRow Height Row HeightdefaultshorttallCustomize ColumnsCustomize ColumnsSelect columns to be displayedNameFI StatusProvinceDistrictCantonVillageFocus AreaStatusStart DateEnd DateRows to display10203050PreviousPage    Of 1Next NameFI StatusProvinceDistrictCantonVillageFocus AreaStatusStart DateEnd DateA1-Tha Luang Village 1 Focus 01 activeChadizaNaviluliNVI_439  A12019-07-312019-06-18A1-Tha Luang Village 1 Focus 01 completeCanton Tha LuangTha Luang VillageTLv1_02  A12019-07-312019-06-18plan104 activeTrat 2Mueng Trat District 2Laem Klat Canton 2Laem Klat Moo 8Tha Sen 8  A12019-06-182019-07-30Rows to display10203050PreviousPage    Of 1Next DefinitionsA1ActiveIndigenous case recorded within the last year.A2Residual Non-ActiveNo indigenous case during the last year, but withing the last 3 years.B1Cleared ReceptiveReceptive area but no indigenous cases within the last 3 years.B2Cleared Non-ReceptiveNon-receptive area."`;
